acceptance-tests/tests/helpers/cmd.js

```diff
@@ -94,67 +86,68 @@ function executeWithInput(processPath, args = [], inputs = [], opts = {}) {
     }
 
     currentInputTimeout = setTimeout(async () => {
-      if (typeof currentInput === 'function') {
-        await currentInput();
+      if (typeof inputs[0] === 'function') {
+        await inputs[0]();
       } else {
-        childProcess.stdin.write(currentInput);
+        childProcess.stdin.write(inputs[0]);
       }
 
-      // Log debug I/O statements on tests
       if (global.config.debug) {
-        console.log('input:', currentInput);
+        console.log('input:', inputs[0]);
       }
 
-      loop(inputs.slice(1));
+      loopInputs(inputs.slice(1));
     }, timeout);
   };
 
-  const promise = new Promise((resolve, reject) => {
-    // Get errors from CLI
-    childProcess.stderr.on('data', data => {
-      // Log debug I/O statements on tests
-      if (global.config.debug) {
-        console.log('error:', data.toString());
-      }
-    });
-
-    // Get output from CLI
-    childProcess.stdout.on('data', data => {
-      // Log debug I/O statements on tests
-      if (global.config.debug) {
-        console.log('output:', data.toString());
-      }
-    });
-
-    childProcess.stderr.once('data', err => {
-      childProcess.stdin.end();
+  // Get errors from CLI for debugging
+  childProcess.stderr.on('data', err => {
+    if (global.config.debug) {
+      console.log('error:', err.toString());
+    }
+  });
 
-      if (currentInputTimeout) {
-        clearTimeout(currentInputTimeout);
-        inputs = [];
-      }
+  // Get output from CLI for debugging
+  childProcess.stdout.on('data', data => {
+    if (global.config.debug) {
+      console.log('output:', data.toString());
+    }
+  });
 
-      // For some reason, when we use 'ora', it is throwing this error,
-      // so for now we won't reject.
-      const blacklistedStrings = ['Loading available API samples'];
+  const promise = new Promise((resolve, reject) => {
+    const handleStderr = err => {
+      // Ignore any allowed errors so tests can continue
+      const allowedErrors = [
+        'Loading available API samples', // When we use 'ora' it is throwing the loading error
+        'DeprecationWarning', // Ignore package deprecation warnings.
+        '[WARNING]', // Ignore our own CLI warning messages
+      ];
 
       const error = err.toString();
-      if (blacklistedStrings.some(s => error.includes(s))) {
+      if (allowedErrors.some(s => error.includes(s))) {
         if (global.config.debug) {
           console.log('suppressed error:', error);
         }
 
+        // Resubscribe if we ignored this error
+        childProcess.stderr.once('data', handleStderr);
```

Comment:
I haven't tested this, but I'm pretty sure we need this because this subscription type is `childProcess.stderr.once()` . Otherwise this subscription goes away after ignoring the first error